local/server.go

```diff
@@ -715,14 +715,16 @@ func (a *apiServer) handleArtifactsUpload(w http.ResponseWriter, r *http.Request
 	}
 	defer file.Close()
 
+	contentDisposition := header.Header.Get("Content-Disposition")
+	filename := filenameRegexp.FindStringSubmatch(contentDisposition)[1]
```

Comment:
What if Content-Disposition doesn't contain `filename=".."`, for whatever reason?

Reply:
Good point. Looks like it's not required by [some](https://www.rfc-editor.org/rfc/rfc1867#section-3.3) [RFCs](https://www.rfc-editor.org/rfc/rfc7578#section-4.2).
It's a "SHOULD" not a "MUST" in the later one.

However, the agent does seem to always provide it. It's a little convoluted why and how, but I think it ultimately because of this method: https://go.dev/src/mime/multipart/writer.go#L140. Obviously, this may change if the implementation of `CreateFormFile` in the standard library changes, this may seem unlikely, but such a change is why I'm writing this code in the first place 🤣 If that happens, it's likely that the [integration test](https://buildkite.com/buildkite/agent/builds/5191#0186217e-33a4-4bd0-9a9f-ecf459643f1e) will fail then as well. However, we do seem to unit test the old implementation [here](https://github.com/buildkite/agent/blob/81cbb6fea466b9f3c810be8260b78389bbd5396e/agent/form_uploader_test.go#L58-L61). So I suggest we change that test to use the same implementation as here, and write a unit test that distinguishes between an absolute and relative upload as well. This would be a PR in the agent repo.

As for what to do here, I'm going to make sure it does not panic if the regex does not match, and I'll surface an error if it's not.

Reply:
> I'm going to make sure it does not panic if the regex does not match, and I'll surface an error if it's not.

Sounds good to me!